.github/workflows/run-application.yaml

```diff
@@ -26,9 +27,22 @@ jobs:
           curl -sSL https://install.python-poetry.org | python3 -
           echo "$HOME/.local/bin" >> $GITHUB_PATH
 
-      - name: Install dependencies and run app
+      - name: Activate environment and install dependencies
         run: |
           make v
           source .venv/bin/activate
           make i
-          tus_storagehandler
+
+      - name: Start server in the background
+        run: |
+          tus_storagehandler &
+          echo $! > server.pid
+
+      - name: Run tests
+        run: |
+          poetry run pytest
```

Comment:
I suppose you only need to run the integration/end-to-end tests, not _all_ tests.